### ISSUE TITLE
[Bug]: Login via Email does not work if Primary Domain or Org Id Scopes are set

### ISSUE THREAD

**Issue by sschoeb** [zitadel/zitadel#6868]
### Preflight Checklist

- [X] I could not find a solution in the documentation, the existing issues or discussions
- [X] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Environment

ZITADEL Cloud

### Describe the problem caused by this bug

Login via E-mail is supposed to work also if adding the domain suffixis enforced. 

This is only partially true currently. The initial problem lies in [checkLoginName](https://github.com/zitadel/zitadel/blob/main/internal/auth/repository/eventsourcing/eventstore/auth_request.go#L738). Whenever either urn:zitadel:iam:org:id:{id} (probably) or urn:zitadel:iam:org:domain:primary:{domainname} (tested) is set the given loginname will always first be suffixed on that line with the given orgdomain, making any e-mail address search unusable. 

The easiest to see this, is that email-login always works on the zitadel console login (no such scopes) but if you go to authorize with one of those scopes it suddenly does not work anymore and you get "User not found" with the identical E-mail. 

I think it should still be able to search by email and the verify that the organisation/domain  is correct. It is very unintuitive when I provide more info what to search that less login ways work for the correct users.

### To reproduce

- Disable the setting "Login mittels E-Mailadresse deaktivieren" in the organization settings
- Try to login with a user, using his email

### Expected behavior

I would expect to be able to login using an email of a user.

### Relevant Configuration

- "Login mittels E-Mailadresse deaktivieren" is NOT active in the org settings

### Additional Context

There is a previous Discord "discussion" about this:
https://discord.com/channels/927474939156643850/1168823849169719297/1168823849169719297

**Comment by MajaEffenberg** [comment]
Hi everyone, any news on this bug?

**Comment by hifabienne** [comment]
> Hi everyone, any news on this bug?

Hei Maja we do not have any news on this right now.
We will update the issue when there will be updates

**Comment by vs-gtadeu** [comment]
FWIW, I can confirm using the Organization ID scope (`urn:zitadel:iam:org:id:{id}`) doesn't prevent using an email address to log in.

I'd also like to leave it documented that having this bug fixed would unblock us from using Zitadel for our username/email address login use case, which consists of having username uniqueness at the organization level without having to expose/require users to provide the username suffix.

**Comment by livio-a** [comment]
OK we rechecked and can state that:
- using the Organization ID works with and without `Add organization domain as suffix to loginnames` option
- using the Organization Primary Domain Scope works only without the option enabled

We will check how to fix that.

@sschoeb can you please recheck if you still experience the issue even with the id scope and if so, send a mail to support with details about the user.

@vs-gtadeu is there a specific reason you need the domain scope or could you also work with the id scope?

**Comment by sschoeb** [comment]
@livio-a Thanks for your feedback. 

If I remove the organization primary domain scope, the email-login does work as expected. But the user is automatically sent to the "instance-login-screen". If he then does want to login with his username, he has to provide the full username including @micromate.ai (which is not needed when having this scope). Our users are now used to just enter the username without the domain, therefore this seems not to be an option to us. 

Long story short:
- I can confirm that it works if only the organization ID is set
- It does not work for us, as we do need the organization primary domain scope, as users are already used to their login names

Hope that helps, let me know if you need some further testing or have any questions on that.

**Comment by livio-a** [comment]
Thanks @sschoeb and @vs-gtadeu for the reply.

I see the point on having both possibilities (username even with suffix and email).
We're working on a fix: https://github.com/zitadel/zitadel/pull/7446.

It should be fixed within next week.
